### ISSUE TITLE
UploadedTO Premium not working.

### ISSUE THREAD

**Issue by Frozen-byte** [pyload/pyload#1866]
I can not provide any  Errorlog messages.
Login is valid, and has good amount of traffic left.

When downloading a file:
`25.09.2015 01:26:15 INFO      HOSTER UploadedTo[914]: Processing as free download...`
Appears and it asks for Captcha.

Any Suggestions how I can provide more informations?
Does anyone have similar issues?

**Comment by GammaC0de** [comment]
could you add this line:
```
                    self.log_debug("self.premium: '%s' self.account.premium:'%s'" % (self.premium, self.account.premium))
```

before line 278:
```
                    if self.premium and (not self.CHECK_TRAFFIC or self.check_traffic_left()):
```
and post the result?

**Comment by GammaC0de** [comment]
could you add this line:
```
                    self.log_debug("self.premium: '%s' self.account.premium:'%s'" % (self.premium, self.account.premium))
```

before line 278:
```
                    if self.premium and (not self.CHECK_TRAFFIC or self.check_traffic_left()):
```
and post the result?

**Comment by mationic** [comment]
Dublicate of https://github.com/pyload/pyload/issues/1853

**Comment by Gutz-Pilz** [comment]
```
25.09.2015 09:41:42 INFO      Download starts: BBTHD.901.rar
25.09.2015 09:41:42 DEBUG     HOSTER UploadedTo[1798]: PROCESS URL http://uploaded.net/file/afxrt0dv | PLUGIN VERSION 0.97
25.09.2015 09:41:42 INFO      HOSTER UploadedTo[1798]: Updating file info...
25.09.2015 09:41:42 DEBUG     HOSTER UploadedTo[1798]: File info: {'status': 2, 'url': u'http://uploaded.net/file/afxrt0dv', 'size': 453440028.0, 'name': 'BBTHD.901.rar', 'pattern': {'ID': u'afxrt0dv'}}
25.09.2015 09:41:42 DEBUG     HOSTER UploadedTo[1798]: Previous file info: {}
25.09.2015 09:41:42 INFO      HOSTER UploadedTo[1798]: File name: BBTHD.901.rar
25.09.2015 09:41:42 INFO      HOSTER UploadedTo[1798]: File size: 453440028 bytes
25.09.2015 09:41:42 INFO      HOSTER UploadedTo[1798]: File status: online
25.09.2015 09:41:42 DEBUG     HOSTER UploadedTo[1798]: LOAD URL http://uploaded.net/file/afxrt0dv | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=False | just_header=False
25.09.2015 09:41:43 DEBUG     HOSTER UploadedTo[1798]: self.premium: 'False' self.account:'False'
25.09.2015 09:41:43 INFO      HOSTER UploadedTo[1798]: Processing as free download...
25.09.2015 09:41:43 DEBUG     HOSTER UploadedTo[1798]: LOAD URL http://uploaded.net/language/en | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=True
25.09.2015 09:41:43 DEBUG     HOSTER UploadedTo[1798]: LOAD URL http://uploaded.net/js/download.js | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 09:41:43 DEBUG     CAPTCHA UploadedTo[1798]: ReCaptcha | Key: 6Lcqz78SAAAAAPgsTYF3UlGf2QFQCNuPMenuyHF3
25.09.2015 09:41:43 DEBUG     HOSTER UploadedTo[1798]: LOAD URL http://www.google.com/recaptcha/api/challenge | cookies=True | get={'k': u'6Lcqz78SAAAAAPgsTYF3UlGf2QFQCNuPMenuyHF3'} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 09:41:43 DEBUG     CAPTCHA UploadedTo[1798]: ReCaptcha | Challenge: 03AHJ_VuuJZ7MwBFVY8UMN4s-G_31lPXLfKacDL7iXVVZEDVK-qaJm7uWbzFHPmrG-z5FauHMhIMVyAakdPjaY3Bfomh94EGCZaQXGctqKjMMksPbh8glWbZip_hVi3rHI7AXnUkIlboC_y-ApHMFByoxBtt0X40e65_DG_UIvD3jMpSD00xdxNGwYHXMsK5OpnbDtjONwDy-B
25.09.2015 09:41:43 DEBUG     HOSTER UploadedTo[1798]: LOAD URL http://www.google.com/recaptcha/api/js/recaptcha.js | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 09:41:43 DEBUG     HOSTER UploadedTo[1798]: LOAD URL http://www.google.com/recaptcha/api/reload | cookies=True | get={'reason': 'i', 'c': u'03AHJ_VuuJZ7MwBFVY8UMN4s-G_31lPXLfKacDL7iXVVZEDVK-qaJm7uWbzFHPmrG-z5FauHMhIMVyAakdPjaY3Bfomh94EGCZaQXGctqKjMMksPbh8glWbZip_hVi3rHI7AXnUkIlboC_y-ApHMFByoxBtt0X40e65_DG_UIvD3jMpSD00xdxNGwYHXMsK5OpnbDtjONwDy-B', 'type': 'image', 'k': u'6Lcqz78SAAAAAPgsTYF3UlGf2QFQCNuPMenuyHF3'} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 09:41:43 DEBUG     CAPTCHA UploadedTo[1798]: ReCaptcha | Second challenge: 03AHJ_Vuv-KzkPHmeurilN-NWBx5StiEpNUwpdReKynnANoR9bvcUrAmJLFqVvF6CAvpRp0rWaYoKqFse_wFYuyner0Odf0aC1jzyNv9HRQETdWF91dh5G8tcdlpXhG0Xlh4q1RIzWbTL2oCB6Dk3crLEeT0UuTmvF_b8GMiFYGxRaTc1bVMZBp9WNXp2jM-oW_K8yZZ0HduXKEfz3IkrMFu1ak-SMZLCGrQ
25.09.2015 09:41:43 DEBUG     CAPTCHA UploadedTo[1798]: ReCaptcha | LOAD URL http://www.google.com/recaptcha/api/image | cookies=True | get={'c': u'03AHJ_Vuv-KzkPHmeurilN-NWBx5StiEpNUwpdReKynnANoR9bvcUrAmJLFqVvF6CAvpRp0rWaYoKqFse_wFYuyner0Odf0aC1jzyNv9HRQETdWF91dh5G8tcdlpXhG0Xlh4q1RIzWbTL2oCB6Dk3crLEeT0UuTmvF_b8GMiFYGxRaTc1bVMZBp9WNXp2jM-oW_K8yZZ0HduXKEfz3IkrMFu1ak-SMZLCGrQ'} | req=None | decode=False | multipart=False | post={} | ref=False | just_header=False


```

**Comment by zhaobb** [comment]
Same here for RapidGatorNet. Account page shows it's Premium but download runs as non-premium users.

Issue happened after pyLoad applied the update for issue #1833.

**Comment by GammaC0de** [comment]
see #1853 and fix #1863

**Comment by Gutz-Pilz** [comment]
theres no class "handle_premium" in .hoster/UploadedTo.py

**Comment by GammaC0de** [comment]
@Gutz-Pilz if you want to support premium downloads for a hoster you must either implement `handle_premium` or specify `LINK_PREMIUM_PATTERN` (as in `UploadedTo.py`) which is a direct download link to the file (in case the hester provides it for premium users)

**Comment by Gutz-Pilz** [comment]
okay. understand. 
thanks a lot

**Comment by GammaC0de** [comment]
:+1:

**Comment by Gutz-Pilz** [comment]
@GammaC0de 
you know something about crypter. i tried my best to fix filecrypt.cc
but totally dont understand the code there.

```
25.09.2015 16:15:52 INFO      Decrypting starts: http://filecrypt.cc/Container/3193DFCBF5.html
25.09.2015 16:15:52 DEBUG     CRYPTER FilecryptCc[1794]: PROCESS URL http://filecrypt.cc/Container/3193DFCBF5.html | PLUGIN VERSION 0.19
25.09.2015 16:15:52 DEBUG     CRYPTER FilecryptCc[1794]: LOAD URL http://filecrypt.cc/Container/3193DFCBF5.html | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 16:15:52 DEBUG     CRYPTER FilecryptCc[1794]: Captcha-URL: ../captcha/captcha.php?namespace=container
25.09.2015 16:15:52 ERROR     Decrypting failed: http://filecrypt.cc/Container/3193DFCBF5.html | global name 'captcha_url' is not defined
Traceback (most recent call last):
  File "/usr/share/pyload/module/PluginThread.py", line 363, in run
    self.active.plugin.preprocessing(self)
  File "/root/.pyload/userplugins/internal/Hoster.py", line 184, in preprocessing
    return self._process(*args, **kwargs)
  File "/root/.pyload/userplugins/internal/Hoster.py", line 179, in _process
    return self.process(self.pyfile)
  File "/root/.pyload/userplugins/internal/Crypter.py", line 39, in process
    self.decrypt(pyfile)
  File "/root/.pyload/userplugins/crypter/FilecryptCc.py", line 52, in decrypt
    self.handle_captcha()
  File "/root/.pyload/userplugins/crypter/FilecryptCc.py", line 95, in handle_captcha
    captcha = self.load(captcha_url, decode=False)
NameError: global name 'captcha_url' is not defined
```

```
25.09.2015 16:24:20 INFO      Added package A World Beyond (2015) containing 1 links
25.09.2015 16:24:21 INFO      Decrypting starts: http://filecrypt.cc/Container/5C4F06651C.html
25.09.2015 16:24:21 DEBUG     CRYPTER FilecryptCc[1803]: PROCESS URL http://filecrypt.cc/Container/5C4F06651C.html | PLUGIN VERSION 0.19
25.09.2015 16:24:21 DEBUG     CRYPTER FilecryptCc[1803]: LOAD URL http://filecrypt.cc/Container/5C4F06651C.html | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 16:24:21 WARNING   CAPTCHA FilecryptCc[1803]: ReCaptcha | Key pattern not found
None
25.09.2015 16:24:21 INFO      CRYPTER FilecryptCc[1803]: No captcha found
25.09.2015 16:24:21 INFO      CRYPTER FilecryptCc[1803]: Found 2 mirrors
25.09.2015 16:24:21 DEBUG     CRYPTER FilecryptCc[1803]: LOAD URL http://filecrypt.cc/Container/5C4F06651C.html?mirror=1 | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 16:24:21 DEBUG     CRYPTER FilecryptCc[1803]: LOAD URL http://filecrypt.cc//Link/eoGaN7oli3yrBXRFmvjP-oRXpH3naqe9BJFrLFAM2OgD0gNL4edN9YshUYWH2L_xFkvHU2EWtcQN6b_5hp4g_DpwQ3w4AzAh8AatA6-mRGbckDyQd_CgV4gzKT57O0T4.html | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 16:24:21 DEBUG     CRYPTER FilecryptCc[1803]: LOAD URL http://filecrypt.cc//Link/9FWfYe89WJV994NdaK7ClSPA6OZyF_SNyHjXdQM_XmTlXIPjoUhESbTu_ewOhBqTFNrbG9WQBNxFaBc5zX3ypQ.html | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 16:24:21 DEBUG     CRYPTER FilecryptCc[1803]: LOAD URL http://filecrypt.cc//Link/6JMeqSqoLIxmQyZq7zj7m7lliqaqqEvHP7xc8_yub-wPQrkapvgCe3FjrFTh_cZTu0oIQhKu9-1hZ5DR9Do0r8OeIK_siXFgYTvnuFHyA6vVineP5IYPui4gOAPhUdBJ.html | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 16:24:21 DEBUG     CRYPTER FilecryptCc[1803]: LOAD URL http://filecrypt.cc//Link/5TZWahj0alL9Bil6GQjbMlwZLhAZNZ62e0wBlDj5snR3GqxCkk3Vs0yWDDBvOrozXHGj2-aDeSvYqh3Mujp59A.html | cookies=True | get={} | req=None | decode=True | multipart=False | post={} | ref=True | just_header=False
25.09.2015 16:24:21 ERROR     Decrypting failed: http://filecrypt.cc/Container/5C4F06651C.html | Decrypt error: No link grabbed | Plugin may be out of date
```

**Comment by GammaC0de** [comment]
@Gutz-Pilz lets continue in #1805

**Comment by Gutz-Pilz** [comment]
kk,
see u there :)

**Comment by tatu007** [comment]
hello,

Same problem with OneFichier.com
```	INFO	HOSTER OneFichierCom[1]: Processing as free download... ```

**Comment by damien599901** [comment]
same problem with RapidGator, 1fichier.com, turbobit, 
Nothing works anymore

**Comment by GammaC0de** [comment]
either wait for @vuolter to merge  fix #1863 or do it manually now
